src/Raven.Server/Documents/Handlers/CountersHandler.cs

```diff
@@ -83,30 +83,33 @@ public ExecuteCounterBatchCommand(DocumentDatabase database)
                 _dictionary = new Dictionary<string, List<CounterOperation>>();
             }
 
-            public void Add(string id, CounterOperation op)
+            public void Add(string id, CounterOperation op, out bool isNew)
             {
+                isNew = false;
                 if (_dictionary.TryGetValue(id, out var existing) == false)
                 {
+                    isNew = true;
                     _dictionary[id] = new List<CounterOperation> { op };
+                    return;
                 }
-
-                else
-                {
-                    existing.Add(op);
-                }
+                existing.Add(op);
             }
 
             protected override int ExecuteCmd(DocumentsOperationContext context)
             {
                 if (_database.ServerStore.Server.Configuration.Core.FeaturesAvailability == FeaturesAvailability.Stable)
                     FeaturesAvailabilityException.Throw("Counters");
 
+                var ops = 0;
+                var countersToAdd = new SortedSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Why do we want to have them sorted?

Reply:
This is a requirement.

Reply:
did you push the code? I'm not seeing this removed?

Reply:
The list of counters was sorted in the previous implementation. I think that we use that when patching a document with a counter.